crates/re_viewer/src/misc/caches/tensor_decode_cache.rs

```diff
@@ -67,7 +106,7 @@ impl DecodeCache {
                             Ok(img) => match Tensor::from_image(img) {
                                 Ok(tensor) => {
                                     if tensor.shape() == maybe_encoded_tensor.shape() {
-                                        Ok(tensor)
+                                        Ok(DecodedTensor(tensor))
```

Comment:
from_image shouldn't ever return a jpeg-encoded tensor, but to make sure we enforce the unreachable guarantee, could make sense for  DecodedTensor to `impl TryFrom<Tensor>` and return Err based on storage.